test/test_python_dispatch.py

```diff
@@ -0,0 +1,139 @@
+import torch
+from torch.testing._internal.common_utils import TestCase, run_tests
+from torch._C import _disabled_torch_function_impl
+
+from typing import Iterator, List, Dict
+import logging
+import contextlib
+
+# TODO: TensorBase should work
+class LoggingTensor(torch.Tensor):
+    elem: torch.Tensor
+
+    @staticmethod
+    def __new__(cls, elem, *args, **kwargs):
+        r = torch.Tensor._make_subclass(cls, elem.to('meta'), elem.requires_grad)
+        r.elem = elem
+        return r
+
+    def __repr__(self):
+        return f"LoggingTensor({self.elem})"
+
+    def __str__(self):
+        return f"LoggingTensor({self.elem})"
+
+    def __format__(self, format_spec):
+        return f"LoggingTensor({self.elem})"
+
+    __torch_function__ = _disabled_torch_function_impl
+
+    @classmethod
+    def __torch_dispatch__(cls, func, types, args=(), kwargs=None):
+        def unwrap(e):
+            if isinstance(e, torch.Tensor):
+                return e.elem
+            elif isinstance(e, (list, tuple)):
+                return [unwrap(sub_e) for sub_e in e]
+            else:
+                return e
```

Comment:
nit: pytrees handle this a bit more succinctly
```
from torch.utils._pytree import tree_map

def unwrap_single(e):
    return e.elem if isinstance(e, torch.Tensor) else e.elem

def wrap_single(e):
    return LoggingTensor(e) if isinstance(e, torch.Tensor) else e

wrap = functools.partial(tree_map, wrap_single)
unwrap = functools.partial(tree_map, unwrap_single)
```

Reply:
oops, forgot this is a thing, pytrees sounds great